external/js/cothority/src/network/connection.ts

```diff
@@ -132,42 +131,85 @@ export class WebSocketConnection implements IConnection {
 /**
  * Multi peer connection that tries all nodes one after another
  */
-export class RosterWSConnection extends WebSocketConnection {
+export class RosterWSConnection {
+    static parallel: number = 2;
     addresses: string[];
+    addressNext: number;
+    connections: WebSocketConnection[] = [];
 
     /**
      * @param r         The roster to use
      * @param service   The name of the service to reach
-     * @param retry = 2 How many times to retry a failing connection with another node
      */
-    constructor(r: Roster, service: string, private retry: number = 2) {
-        super("", service);
+    constructor(r: Roster, service: string) {
         this.addresses = r.list.map((conode) => conode.getWebSocketAddress());
         shuffle(this.addresses);
+        this.addressNext = RosterWSConnection.parallel;
```

Comment:
Please let that be parameterized so we can disable it if necessary.

Reply:
If you put `parallel = 1`, it will only do one connection at the time, and still rotate through the nodes if that connection fails or times out.

Another solution would be to add a new `RosterWSParallelConnection` and then use that if the user wants it.

Reply:
No no that's fine, that's what I expect. You may want to have only one connection at a time so even if most of the time it will use the default value, it's always nice to have the choice.

Reply:
So instead of `static parallel: number = 2;` you want to have `setParallelCalls` as a method?

Reply:
It can be a parameter in the constructor.